src/Cardano/Transaction.h

```diff
@@ -0,0 +1,72 @@
+// Copyright © 2017-2022 Trust Wallet.
+//
+// This file is part of Trust. The full Trust copyright notice, including
+// terms governing use, modification, and redistribution, is contained in the
+// file LICENSE at the root of the source code distribution tree.
+
+#pragma once
+
+#include "AddressV3.h"
+
+#include "Data.h"
+#include "../proto/Cardano.pb.h"
+#include "../proto/Common.pb.h"
+
+#include <vector>
+
+namespace TW::Cardano {
+
+typedef uint64_t Amount;
+
+class OutPoint {
+public:
+    Data txHash;
+    uint64_t outputIndex;
+
+    static OutPoint fromProto(const Proto::OutPoint& proto);
+};
+
+class TxInput: public OutPoint {
+public:
+    std::string address;
+
+    /// ADA amount
+    Amount amount;
+
+    static TxInput fromProto(const Proto::TxInput& proto);
+};
+
+class TxOutput {
+public:
+    Data address;
+    /// ADA amount
+    Amount amount;
```

Comment:
only support transfer ada coin, if some input hash container other token, you don't write other token amount in output, there will be a【ValueNotConservedUTxO】 error.

Reply:
Added support for tokens in the proto file, and explicit check and error in this case.  To be handled later.